src/pytest_inmanta_lsm/managed_service_instance.py

```diff
@@ -321,7 +321,7 @@ def compare_states(current_state: State, wait_for_states: List[str]) -> bool:
                     # Version is not given, so version does not need to be verified
                     return True
                 elif current_state.version not in desired_versions:
-                    raise VersionMismatchError(self, desired_versions, current_state.version)
+                    raise VersionMismatchError(self, desired_versions, current_state.version or 0)
```

Comment:
Do you know why `current_state.version` is Optional? What is the semantic meaning of `current_state.version=None`. When I look at the code it seems like `current_state.version is None` can never happen.

Reply:
I think this is because in the version comparison mechanism, None means whatever value the other has is a match of mine.  I simply did it to make mypy happier, but I will remove it